### PR TITLE
Rem 448

### PR DESCRIPTION
**Pull request by omarZrayka23**
API for second tab = IV.MTS
In attachment, red fields are the mandatory ones.

In first tab:

-reference has maxLength=3

-purity & reportingPurity fields have minimum value = 0.001, maximum value = 1 and shows only 5 decimals

In second tab:

-laborValuePerGram shows 2 decimals only

-purity has minimum value = 0.001, maximum value = 1 and shows 5 decimals

### CODE REVIEW

**Comment by Ghinwa-Arab** [comment]
image

returns no data

**Comment by Ghinwa-Arab** [comment]
image

1. sku should be lookup not combo
2. minimize dropdown width

**Comment by Ghinwa-Arab** [comment]
image
(labor value per gram) in inline grid shouldn't have maximum = 1

**Comment by Ghinwa-Arab** [comment]
image

handle this on save (send correct json if there's no items)

**Comment by Ghinwa-Arab** [comment]
image

sku is mandatory field

**Comment by Ghinwa-Arab** [comment]
image

purity not mandatory

**Comment by Ghinwa-Arab** [comment]
image

currency not saving

**Comment by Ghinwa-Arab** [comment]
image

purity in edit mode should be readonly

File: src/pages/iv-metals/Window/MetalsWindow.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// ** Custom Imports
```
Comment (Ghinwa-Arab) [inline src/pages/iv-metals/Window/MetalsWindow.js:1]: remove

File: src/pages/iv-metals/index.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import { ControlContext } from 'src/providers/ControlContext'
+
+const TaxCodes = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const { stack } = useWindow()
+
+  async function fetchGridData() {
+    const response = await getRequest({
```
Comment (Ghinwa-Arab) [inline src/pages/iv-metals/index.js:23]: no need for assigning api response to const u can directly return it 
` return await getRequest({
      extension: InventoryRepository.Metals.qry,
      parameters: `_filter=`
    })
`

File: src/pages/iv-metals/index.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+  const {
+    query: { data },
+    labels: _labels,
+    invalidate,
+    refetch,
+    access
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: InventoryRepository.Metals.page,
```
Comment (Ghinwa-Arab) [inline src/pages/iv-metals/index.js:39]: here u are using page endpoint while above in fetchGridData u are using qry
let them call same endpoint

File: src/pages/iv-metals/forms/MetalsForm.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { MasterSource } from 'src/resources/MasterSource'
```
Comment (Ghinwa-Arab) [inline src/pages/iv-metals/forms/MetalsForm.js:13]: remove MasterSource, not used anywhere

File: src/pages/iv-metals/forms/MetalsForm.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+          extension: InventoryRepository.Metals.set,
+          record: JSON.stringify(obj)
+        })
+        if (!obj.recordId) {
+          setStore(prevStore => ({
+            ...prevStore,
+            recordId: response.recordId
+          }))
+          toast.success(platformLabels.Added)
+          formik.setValues({
```
Comment (Ghinwa-Arab) [inline src/pages/iv-metals/forms/MetalsForm.js:60]: replace with `formik.setFieldValue('recordid',response.recordId)`

File: src/pages/iv-metals/forms/MetalsForm.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+          formik.setValues(res.record)
+        }
+      } catch (exception) {}
+    })()
+  }, [])
+
+  return (
+    <FormShell
+      resourceId={ResourceIds.TaxCodes}
```
Comment (Ghinwa-Arab) [inline src/pages/iv-metals/forms/MetalsForm.js:90]: fix resourceId

File: src/pages/iv-metals/forms/MetalsForm.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+  const { recordId } = store
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: InventoryRepository.Metals.page
+  })
+
+  const { formik } = useForm({
+    initialValues: { recordId: store.recordId, name: '', reference: '', purity: '', reportingPurity: '' },
```
Comment (Ghinwa-Arab) [inline src/pages/iv-metals/forms/MetalsForm.js:30]: initialValues
1. shouldn't contain name
2. missing currencyId

File: src/pages/iv-metals/forms/ScrapForm.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+      items: items
+    }
+
+    postRequest({
+      extension: InventoryRepository.Scrap.set2,
+      record: JSON.stringify(data)
+    })
+      .then(res => {
+        toast.success(platformLabels.Added)
+        setStore(prevStore => ({
```
Comment (Ghinwa-Arab) [inline src/pages/iv-metals/forms/ScrapForm.js:78]: Could you please clarify why we're updating items in the store when it only contains recordId? If it's not necessary, please remove it, as well as from the props and any references in the window.

File: src/pages/iv-metals/forms/ScrapForm.js [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+        })
+        .catch(error => {})
+    }
+  }, [])
+
+  return (
+    <>
+      <FormShell
+        form={formik}
+        resourceId={ResourceIds.TaxCodes}
```
Comment (Ghinwa-Arab) [inline src/pages/iv-metals/forms/ScrapForm.js:110]: wrong resourceId

File: src/repositories/InventoryRepository.js [hunk 10]
```diff
@@ -67,5 +67,18 @@ export const InventoryRepository = {
...
+  },
+  Metals: {
+    qry: service + 'qryMTL',
+    get: service + 'getMTL',
+    set: service + 'setMTL',
+    del: service + 'delMTL',
+    snapshot: service + 'snapshotMTL'
+  },
+  Scrap: {
+    qry: service + 'qryMTS',
```
Comment (Ghinwa-Arab) [inline src/repositories/InventoryRepository.js:79]: remove del and get if not used

File: src/repositories/InventoryRepository.js [hunk 11]
```diff
@@ -67,5 +67,18 @@ export const InventoryRepository = {
     get: service + 'getDI',
     set: service + 'setDI',
     del: service + 'delDI'
+  },
+  Metals: {
+    qry: service + 'qryMTL',
```
Comment (Ghinwa-Arab) [inline src/repositories/InventoryRepository.js:72]: add page endpoint and remove qry, if snapshot not used remove it

File: src/pages/iv-metals/forms/ScrapForm.js [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+                  props: {
+                    maxLength: 6,
+                    decimalScale: 2
+                  }
+                },
+                {
+                  component: 'numberfield',
+                  label: labels.amount,
+                  name: 'purity',
+                  props: {
```
Comment (Alinadimmokdad) [inline src/pages/iv-metals/forms/ScrapForm.js:160]: why label.amount

File: src/pages/iv-metals/forms/ScrapForm.js [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+      scrap: yup
+        .array()
+        .of(
+          yup.object().shape({
+            sku: yup.string().required(),
+            purity: yup
+              .number()
+              .nullable()
+              .test('is-valid-reportingPurity', function (value) {
+                return !value || (value < 0.001 && value > 1) ? true : false
```
Comment (Alinadimmokdad) [inline src/pages/iv-metals/forms/ScrapForm.js:33]: Screenshot 2024-09-10 150126

**Review by Alinadimmokdad — APPROVED** [review]

**Pull request merged**